src/cpp/src/tokenizer.cpp

```diff
@@ -59,6 +59,19 @@ constexpr char bos_token_key_name[] = "bos_token";
 constexpr char eos_token_key_name[] = "eos_token";
 constexpr char pad_token_key_name[] = "pad_token";
 
+static ov::Core get_core_singleton() {
+    static ov::Core core;
+    static bool is_extension_added = false;
```

Comment:
I never tried it, but I got an idea. Since static var is initialized only once, you can define a function returning ov::Core: `static ov::Core core = core_with_extension();`. That function will probably be called only once.